sdk/daml-lf/validation/BUILD.bazel

```diff
@@ -34,6 +38,33 @@ da_scala_library(
     ],
 )
 
+da_scala_library(
+    name = "upgrade-check-main",
+    srcs = glob(["src/main/**/validation/UpgradeCheckMain.scala"]),
+    scala_deps = [
+        "@maven//:org_scalaz_scalaz_core",
+        "@maven//:org_typelevel_cats_core",
+        "@maven//:org_typelevel_cats_kernel",
+    ],
+    scalacopts = lf_scalacopts_stricter,
+    unused_dependency_checker_mode = "off",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//canton:community_ledger_ledger-api-core",
+        "//canton:community_ledger_ledger-common",
+        "//canton:community_util-logging",
+        "//canton:daml-common-staging_daml-errors",
```

Comment:
We are still depending on canton libraries :(. 
It may be a good tack for future gardening.

Reply:
Yes, sadly the scala validation code lives in Canton